### ISSUE TITLE
Add totalsize metric in MongoDB collstats metricset

### ISSUE THREAD

**Issue by ishleenk17** [elastic/beats#42003]
**Describe the enhancement:**

Add the [totalsize](https://www.mongodb.com/docs/v4.4/reference/method/db.collection.totalSize/#:~:text=To%20use%20the%20database%20command,every%20index%20on%20the%20collection) metric to mongodb collstats.
This is an important metric to report the total size of the collection. 

Once this is added to the beats module, we should add the same to Integrations datastream as well. 

```[tasklist]
## Tasks
- [ ] https://github.com/elastic/beats/pull/42171
- [ ] https://github.com/elastic/integrations/issues/12406
```

**Comment by ishleenk17** [comment]
We should revisit all the collstats metrics as I think we should add couple other important metrics as well. 
The list of the metrics [available](https://www.mongodb.com/docs/v4.4/reference/method/js-collection/) .

**Comment by stefans-elastic** [comment]
This confuses me a bit: it looks like `collstats` module  doesn't actually return `collstats` data. It return results of `top` command: https://github.com/elastic/beats/blob/main/metricbeat/module/mongodb/collstats/collstats.go#L79
Also if we are going to fetch `collstats` metrics I think we need to call `collstats`  command on each collection (which is likely going to degrade performance of the module). Are we ok with this?

**Comment by ishleenk17** [comment]
That's right, this is what I noticed as well while checking the beats code. That we are returning just the usage metrics for the collections from the top command. Listed [here](https://www.elastic.co/guide/en/integrations/current/mongodb.html#mongodb-collstats) as well. We are not having any other metrics. Hence I pointed out we might want to revisit the complete collstats module. 

> Also if we are going to fetch collstats metrics I think we need to call collstats command on each collection (which is likely going to degrade performance of the module)

I think we don't have any way other than using the collstats command, is that right ?

**Comment by stefans-elastic** [comment]
> I think we don't have any way other than using the collstats command, is that right ?

That's my understanding (I haven't found any other way)

**Comment by stefans-elastic** [comment]
@ishleenk17 I've come up with a "PoC" (kinda): https://github.com/elastic/beats/commit/eb7bbcb09d59ba76426551163a3e5e986fa1b39f

If this solution makes sense I will proceed and expand it with other `collStats` fields

**Comment by ishleenk17** [comment]
This looks fine. Thanks Stefan.
Let's come up with the most relevant list of metrics that we would like to include for collstats.

**Comment by stefans-elastic** [comment]
I suggest:
- [size](size)
-  [count](https://www.mongodb.com/docs/v4.4/reference/command/collStats/#mongodb-data-collStats.count)
- [avgObjSize](https://www.mongodb.com/docs/v4.4/reference/command/collStats/#mongodb-data-collStats.avgObjSize)
- [storageSize](https://www.mongodb.com/docs/v4.4/reference/command/collStats/#mongodb-data-collStats.storageSize)
- [totalIndexSize](https://www.mongodb.com/docs/v4.4/reference/command/collStats/#mongodb-data-collStats.totalIndexSize)
- [totalSize](https://www.mongodb.com/docs/v4.4/reference/command/collStats/#mongodb-data-collStats.totalSize)

Please let me know what do you think.

P.S. I've [updated](https://github.com/stefans-elastic/beats/commit/9ae12588d2d1109407aedf510ffcd612957475ec) the PoC a bit (introduced `WaitGroup`'s). I think it sould make the performance better.

**Comment by ishleenk17** [comment]
The list looks good. Lets also add below two to these

[max](https://www.mongodb.com/docs/v4.4/reference/command/collStats/#mongodb-data-collStats.max)
[nindexes](https://www.mongodb.com/docs/v4.4/reference/command/collStats/#mongodb-data-collStats.nindexes)

**Comment by stefans-elastic** [comment]
@ishleenk17 done and PR is ready for review - https://github.com/elastic/beats/pull/42171

**Comment by stefans-elastic** [comment]
Note: there is CI issue with [backport PR ](https://github.com/elastic/beats/pull/42358) for this feature. The problem occurs when CI tries to download one of the dependencies but the site doesn't exists anymore so it fails. The problem is caused by go.opencensus.io (which is brought by github.com/dgraph-io/badger.

There should be a quick fix available shortly - it will prepopulate dependencies on CI VM (so CI doesn't have to download them). PR for quick fix is https://github.com/elastic/ci-agent-images/pull/1187. For long term solution there is issue open for getting rid of github.com/dgraph-io/badger - https://github.com/elastic/beats/issues/42443.

**Comment by stefans-elastic** [comment]
both [beats](https://github.com/elastic/beats/pull/42171) and [integrations](https://github.com/elastic/integrations/pull/12407) parts are merged
